checks.d/marathon.py

```diff
@@ -20,98 +28,37 @@ def check(self, instance):
-        # Use a hash of the URL as an aggregation key
-        aggregation_key = md5(url).hexdigest()
-        tags = ['url:%s' % url]
-        msg = None
-        status = None
+    def get_json(self, url, timeout):
         try:
-            r = requests.get(url + "/v2/apps", timeout=timeout)
-            if r.status_code != 200:
-                self.status_code_event(url, r, aggregation_key)
-                status = AgentCheck.CRITICAL
-                msg = "Got %s when hitting %s" % (r.status_code, url)
+            r = requests.get(url, timeout=timeout)
         except requests.exceptions.Timeout:
             # If there's a timeout
-            self.timeout_event(url, timeout, aggregation_key)
-            msg = "%s seconds timeout when hitting %s" % (timeout, url)
-            status = AgentCheck.CRITICAL
-        except Exception as e:
-            msg = e.message
-            status = AgentCheck.CRITICAL
-        finally:
-            if status is AgentCheck.CRITICAL:
-                self.service_check(self.SERVICE_CHECK_NAME, status, tags=tags, message=msg)
-                raise Exception(msg)
-
-        req_json = r.json()
-        app_count = len(req_json['apps'])
-
-        if app_count is 0:
-            status = AgentCheck.WARN
-            msg = "No marathon applications detected at %s" % url
-        else:
-            instance_count = 0
-            for app in req_json['apps']:
-                instance_count += app['instances']
-            status = AgentCheck.CRITICAL if instance_count is 0 else AgentCheck.OK
-            msg = "%s Marathon app(s) detected with %s instances running at %s" % (app_count, instance_count, url)
-
-
-        self.service_check(self.SERVICE_CHECK_NAME, status, tags=tags, message=msg)
-
-        return req_json
-
-    def get_v2_app_versions(self, url, app_id, timeout):
-        # Use a hash of the URL as an aggregation key
-        aggregation_key = md5(url).hexdigest()
-
-        try:
-            r = requests.get(url + "/v2/apps/" + app_id + "/versions", timeout=timeout)
-        except requests.exceptions.Timeout:
-            # If there's a timeout
-            self.timeout_event(url, timeout, aggregation_key)
-            self.warning("Timeout when hitting %s" % url)
-            return None
+            self.service_check(self.SERVICE_CHECK_NAME, AgentCheck.CRITICAL,
+                message='%s timed out after %s seconds.' % (url, timeout),
+                tags = ["url:{}".format(url)])
+            raise Exception("Timeout when hitting %s" % url)
 
         if r.status_code != 200:
-            self.status_code_event(url, r, aggregation_key)
-            self.warning("Got %s when hitting %s" % (r.status_code, url))
+            self.status_code_event(url, r)
+            raise Exception("Got %s when hitting %s" % (r.status_code, url))
+        else:
+            self.service_check(self.SERVICE_CHECK_NAME, AgentCheck.OK,
+                tags = ["url:{}".format(url)]
```

Comment:
Yes, sorry! I meant to ask where you'd end up with multiple URLs and not want the whole thing. I'm assuming anyone using this wold be hitting something on localhost, which means you'd differentiate by either port or path. Where would trimming down to the base url be useful?